riptide-core/src/main/java/org/zalando/riptide/SerializationPlugin.java

```diff
@@ -0,0 +1,40 @@
+package org.zalando.riptide;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpOutputMessage;
+
+import java.io.ByteArrayOutputStream;
+
+import static org.springframework.util.CollectionUtils.toMultiValueMap;
+
+@AllArgsConstructor
+final class SerializationPlugin implements Plugin {
+
+    private final MessageWriter writer;
+
+    @Override
+    public RequestExecution aroundSerialization(final RequestExecution execution) {
+        return arguments -> {
+            final HttpHeaders headers = new HttpHeaders();
+            headers.addAll(toMultiValueMap(arguments.getHeaders()));
+            final ByteArrayOutputStream stream = new ByteArrayOutputStream();
+            final HttpOutputMessage output = new CapturingHttpOutputMessage(headers, stream);
+
+            writer.write(output, arguments);
+
+            return execution.execute(arguments
+                    .replaceHeaders(headers)
+                    .withEntity(stream.toByteArray()));
```

Comment:
I wonder how often we now copy the request body. Two times or three times?

Reply:
This should be the first time within Riptide now. I'm thinking about using an `OutputStream` instead but it's certainly harder to use then.

Reply:
Logbook would do another copy, unless we implement it natively within Riptide #604 and re-use the copy of the entity.